backend-rust/src/graphql_api/transaction_metrics.rs

```diff
@@ -0,0 +1,119 @@
+use std::sync::Arc;
+
+use async_graphql::{Context, Object, SimpleObject};
+use sqlx::postgres::types::PgInterval;
+
+use crate::graphql_api::{get_pool, ApiError, ApiResult, DateTime, MetricsPeriod, TimeSpan};
+
+#[derive(Default)]
+pub(crate) struct TransactionMetricsQuery;
+
+#[derive(SimpleObject)]
+struct TransactionMetrics {
+    /// Total number of transactions (all time).
+    last_cumulative_transaction_count: i64,
+    /// Total number of transactions in the requested period.
+    transaction_count: i64,
+    buckets: TransactionMetricsBuckets,
+}
+
+#[derive(SimpleObject)]
+struct TransactionMetricsBuckets {
+    /// The width (time interval) of each bucket.
+    bucket_width: TimeSpan,
```

Comment:
A general comment regarding the number of buckets: It seems to be hardcoded to 31 in ccdscan codebase. It would be better if the number of buckets is tracked here and could be general so the front-end has a choice how many buckets to display.

Reply:
Hmm I'm not sure that makes sense. Each period choice has a set bucket width. I.e. if you choose 7 days, each bucket is 6 hours. That will always determine a set number of buckets. I guess we could change it so you could choose *both* a period and a bucket width separately, but I don't think that would be too useful.

Reply:
For now, let's keep it as it is, so we don't have to adjust the front-end. I just find it weird that we are fixed to display always `31` buckets on the front-end.